tests/Test/Xrefcheck/TooManyRequestsSpec.hs

```diff
@@ -0,0 +1,142 @@
+
+import Xrefcheck.Config
+import Xrefcheck.Core
+import Xrefcheck.Progress
+import Xrefcheck.Util
+import Xrefcheck.Verify
+
+spec :: Spec
+spec = do
+  describe "429 response tests" $ do
+    it "Returns 200 eventually" $ do
+      let prog = Progress{ pTotal = 1
+                         , pCurrent = 1
+                         , pErrorsUnfixable = 0
+                         , pErrorsFixable = 0
+                         , pTaskTimestamp = Nothing
+                         }
+      checkLinkAndProgressWithServer (mock429 ok200)
+        "http://127.0.0.1:5000/429" prog $ VerifyResult []
+    it "Returns 503 eventually" $ do
+      let prog = Progress{ pTotal = 1
+                         , pCurrent = 1
+                         , pErrorsUnfixable = 1
+                         , pErrorsFixable = 0
+                         , pTaskTimestamp = Nothing
+                         }
+      checkLinkAndProgressWithServer (mock429 serviceUnavailable503)
+        "http://127.0.0.1:5000/429" prog $ VerifyResult
+          [ ExternalHttpResourceUnavailable $
+              Status { statusCode = 503, statusMessage = "Service Unavailable"}
+          ]
+    it "Successfully updates the new retry-after value" $ do
+      E.bracket (forkIO (newIORef 0 >>= mock429 ok200)) killThread $ \_ -> do
+        now <- getPOSIXTime <&> posixTimeToTimeSecond
+        progressRef <- newIORef VerifyProgress
+              { vrLocal = initProgress 0
+              , vrExternal = Progress
+                  { pTotal = 2
+                  , pCurrent = 1
+                  , pErrorsUnfixable = 0
+                  , pErrorsFixable = 0
+                  , pTaskTimestamp = Just (TaskTimestamp (sec 3) (now -:- sec 1.5))
+                  }
+              }
+        _ <- verifyReferenceDef''
+          (Reference "" "http://127.0.0.1:5000/429retryafterupdate" Nothing (Position Nothing))
+          progressRef
+        Progress{..} <- vrExternal <$> readIORef progressRef
+        let ttc = ttTimeToCompletion <$> pTaskTimestamp
+        expectationTrial (ttc == Just (sec 2)) $
+          "Expected time to completion be equal to " ++ show (Just $ sec 2) ++
+          ", but instead it's " ++ show ttc
+  where
+    checkLinkAndProgressWithServer mock link progress vrExpectation =
+      E.bracket (forkIO (newIORef 0 >>= mock)) killThread $ \_ -> do
+        (result, progRes) <- verifyReferenceDef' link
+        expectationTrial (result == vrExpectation) . pretty $ unlinesF
+          [ "Verification results differ: expected"
+          , indentF 2 (show vrExpectation)
+          , "but got"
+          , indentF 2 (show result)
+          ]
+        expectationTrial (progRes `progEquiv` progress) . pretty $ unlinesF
+          [ "Expected the progress bar state to be"
+          , indentF 2 (show progress)
+          , "but got"
+          , indentF 2 (show progRes)
+          ]
+      where
+        progEquiv :: Eq a => Progress a -> Progress a -> Bool
```

Comment:
Maybe add a comment here explaining that the purpose of this function is to check every field _except_ `pTaskTimestamp` and, importantly, why we do not want to check this field.